CMakeLists.txt

```diff
@@ -38,8 +38,14 @@ if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release")
 endif()
 
+option(USE_RELATIVE_LIB_PATH "Use relative path to libear")
+
 set(EAR_LIB_FILE ${CMAKE_SHARED_LIBRARY_PREFIX}ear${CMAKE_SHARED_LIBRARY_SUFFIX})
-set(DEFAULT_PRELOAD_FILE ${CMAKE_INSTALL_FULL_LIBDIR}/${EAR_LIB_FILE})
+if (USE_RELATIVE_LIB_PATH)
```

Comment:
I like the idea of using only the library name in LD_PRELOAD. But then LD_LBRARY_PATH shall contain the path to the libear file (or files). As far as I can recall this was suggested earlier when people were suffered by multilib builds (32 and 64 bit architecture mix).
So, what do you think, would it better to no choice to use relative or absolute path, but use the always relative and maintain the LD_LIBRARY_PATH environment?